skopt/optimizer/optimizer.py

```diff
@@ -446,8 +446,11 @@ def _ask(self):
             min_delta_x = min([self.space.distance(next_x, xi)
                                for xi in self.Xi])
             if abs(min_delta_x) <= 1e-8:
```

Comment:
> Ideally, the next best point should be evaluated instead, but I do not know how to do that.

I guess you would have to move this check to where `self._next_x` is _set_ and recompute it there.